octobot_trading/exchanges/implementations/ccxt_exchange_commons.py

```diff
@@ -0,0 +1,285 @@
+# pylint: disable=E0611
+#  Drakkar-Software OctoBot-Trading
+#  Copyright (c) Drakkar-Software, All rights reserved.
+#
+#  This library is free software; you can redistribute it and/or
+#  modify it under the terms of the GNU Lesser General Public
+#  License as published by the Free Software Foundation; either
+#  version 3.0 of the License, or (at your option) any later version.
+#
+#  This library is distributed in the hope that it will be useful,
+#  but WITHOUT ANY WARRANTY; without even the implied warranty of
+#  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+#  Lesser General Public License for more details.
+#
+#  You should have received a copy of the GNU Lesser General Public
+#  License along with this library.
+import decimal
+import typing
+
+from octobot_commons import enums as common_enums
+
+import octobot_trading.enums as enums
+from octobot_trading.exchanges.config import (
+    ccxt_exchange_settings,
+    ccxt_exchange_ui_settings,
+)
+import octobot_trading.exchanges.types as exchanges_types
+import octobot_trading.exchanges.connectors as exchange_connectors
+
+
+class CCXTExchangeCommons(exchanges_types.SpotExchange):
+    CONNECTOR_CLASS = exchange_connectors.CCXTExchange
+    CONNECTOR_CONFIG: ccxt_exchange_settings.CCXTExchangeConfig = (
+        ccxt_exchange_settings.CCXTExchangeConfig
+    )
+
+    def __init__(self, config, exchange_manager):
+        self.initialize_connector_config()
+        super().__init__(config, exchange_manager)
+        self.connector = self.CONNECTOR_CLASS(
+            config,
+            exchange_manager,
+            additional_ccxt_config=self.get_additional_connector_config(),
+            connector_config=self.CONNECTOR_CONFIG,
+        )
+
+        self.connector.client.options["defaultType"] = self.get_default_type()
+
+    @classmethod
+    def initialize_connector_config(cls):
+        try:
+            cls.CONNECTOR_CONFIG = cls.CONNECTOR_CONFIG(cls.CONNECTOR_CLASS)
+        except TypeError:
+            pass  # already initialized
+
+    async def initialize_impl(self):
+        await self.connector.initialize()
+        self.symbols = self.connector.symbols
+        self.time_frames = self.connector.time_frames
+
+    @classmethod
+    def init_user_inputs(cls, inputs: dict) -> None:
+        """
+        Called at constructor, should define all the exchange's user inputs.
+        """
+        if not cls.CONNECTOR_CONFIG.is_fully_tested_and_supported():
```

Comment:
I don't think we should enable it for anyone. I think we should add something like checking if the mode is in dev mode

Reply:
Ok :smile: 
The most important thing for me is that users can't access this if they don't try to access it. Because it really is a very specific feature.

Reply:
In my opinion we should add a check to ensure we are using dev mode to prevent any user to access it without wanting to.
We can add a doc that explains how to setup the experimental mode in https://exchanges.octobot.info/

Reply:
can dev mode be used with any build? Also with the exe?
It would be nice to add the dev mode option to the about page. so any user can try it